### ISSUE TITLE
Several unit tests failing

### ISSUE THREAD

**Issue by johlo** [mysql-otp/mysql-otp#43]
@zuiderkwast As I mentioned in my previous PR several unit tests are failing for me.
I'm running MySQL 5.7.13 (installed via brew) on Mac OS X 10.11.4.
Here's the full output from `make tests` on a pristine `master` checkout:

```
$ make tests
 APP    mysql.app.src
 GEN    test-dir
 GEN    eunit
mysql_tests: query_test_ (Text protocol)...*failed*
in function mysql_tests:text_protocol/1 (test/mysql_tests.erl, line 278)
**error:{badmatch,{error,{1364,<<"HY000">>,
                  <<"Field 'tx' doesn't have a default value">>}}}
  output:<<"">>

mysql_tests: query_test_ (Binary protocol)...*failed*
in function mysql_tests:binary_protocol/1 (test/mysql_tests.erl, line 298)
**error:{badmatch,{error,{1050,<<"42S01">>,<<"Table 't' already exists">>}}}
  output:<<"">>

mysql_tests: query_test_ (INT)...*failed*
in function mysql_tests:write_read_text_binary/5 (test/mysql_tests.erl, line 527)
in call from mysql_tests:int/1 (test/mysql_tests.erl, line 412)
**error:{badmatch,{error,{1264,<<"22003">>,
                  <<"Out of range value for column 'i' at row 1">>}}}
  output:<<"">>

mysql_tests: query_test_ (DATE)...*failed*
in function mysql_tests:write_read_text_binary/5 (test/mysql_tests.erl, line 527)
in call from lists:foreach/2 (lists.erl, line 1337)
in call from mysql_tests:date/1 (test/mysql_tests.erl, line 439)
**error:{badmatch,{error,{1292,<<"22007">>,
                  <<"Incorrect date value: '0000-00-00' for column 'd"...>>}}}
  output:<<"">>

mysql_tests: query_test_ (DATETIME)...*failed*
in function mysql_tests:write_read_text_binary/5 (test/mysql_tests.erl, line 527)
in call from lists:foreach/2 (lists.erl, line 1337)
in call from mysql_tests:datetime/1 (test/mysql_tests.erl, line 473)
**error:{badmatch,{error,{1292,<<"22007">>,
                  <<"Incorrect datetime value: '0000-00-00 00:00:00' "...>>}}}
  output:<<"">>

mysql_tests: query_test_ (Microseconds)...*failed*
in function mysql_tests:test_datetime_microseconds/1 (test/mysql_tests.erl, line 512)
**error:{badmatch,{error,{1050,<<"42S01">>,<<"Table 'dt' already exists">>}}}
  output:<<"">>

mysql_tests: log_warnings_test...*failed*
in function mysql_tests:log_warnings_test/0 (test/mysql_tests.erl, line 192)
**error:{badmatch,{error,{badmatch,{error,{1364,<<"HY000">>,
                                   <<"Field 'x' doesn't have a default"...>>}}},
                 [{mysql_tests,'-log_warnings_test/0-fun-0-',1,
                               [{file,"test/mysql_tests.erl"},{line,193}]},
                  {error_logger_acc,capture,1,
                                    [{file,"test/error_logger_acc.erl"},
                                     {line,28}]},
                  {mysql_tests,log_warnings_test,0,
                               [{file,"test/mysql_tests.erl"},{line,192}]},
                  {eunit_test,'-mf_wrapper/2-fun-0-',2,
                              [{file,"eunit_test.erl"},{line,266}]},
                  {eunit_test,run_testfun,1,
                              [{file,"eunit_test.erl"},{line,65}]},
                  {eunit_proc,run_test,1,[{file,"eunit_proc.erl"},{line,503}]},
                  {eunit_proc,with_timeout,3,[{file,[...]},{line,...}]},
                  {eunit_proc,handle_test,2,[{file,...},{...}]}],
                 []}}
  output:<<"">>

=======================================================
  Failed: 7.  Skipped: 0.  Passed: 51.
make: *** [eunit] Error 1
```

Some errors (maybe all?) seem related to `sql_mode` settings. The eunit tests assume that it will get warnings for e.g. missing values on insert, but an error is issued. The mysql behaviour for these cases are controlled by `sql_mode` settings.
These are my settings: (query: `SELECT @@GLOBAL.sql_mode;`)
```
ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
```
Those are the settings I got when installing mysql, I don't know if different distros can have different settings or if it is defined by the mysql version. But maybe it's a good idea for the unit tests to explicitly set the sql_mode variable on a session basis to get the expected behaviour?

**Comment by zuiderkwast** [comment]
I guess the solution is to set the session variable sql_mode to this in the tests.

My settings (`SELECT @@GLOBAL.sql_mode;`) is only `NO_ENGINE_SUBSTITUTION`. With this the tests pass.

**Comment by johlo** [comment]
I did a quick test adding a `SET SESSION sql_mode = "NO_ENGINE_SUBSTITUTION,NO_AUTO_CREATE_USER"` query when starting the session in mysql_tests.erl, it works. I needed to add `NO_AUTO_CREATE_USER` to avoid some deprecation warning from mysql.

Maybe doing this is good enough, if you think so I can add a PR with the change.
 As an alternative I was thinking that we could have an `sql_mode` option to the `mysql:start_link` function which would set the sql_mode for the session. But I'm not sure it makes sense, might be confusing.

**Comment by johlo** [comment]
ping @zuiderkwast

**Comment by zuiderkwast** [comment]
Sorry for the delay. Setting sql_mode in the failing tests would be sufficient I think. There is already a was to run queries on connect. I don't think we need a specialized option for sql_mode. A PR is most welcome!

**Comment by johlo** [comment]
@zuiderkwast Opened PR #45 now, please a have a look!

**Comment by zuiderkwast** [comment]
Solved in PR #45. Thx!
